activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/runtime/ProcessInstanceQueryTest.java

```diff
@@ -1836,4 +1836,22 @@ public void testQueryStartedBy() throws Exception {
 
     assertThat(processInstances).hasSize(1);
   }
+
+  @Deployment(resources = { "org/activiti/engine/test/api/oneTaskProcess.bpmn20.xml" })
+  public void testOrQueryByStartedBy() throws Exception {
+    final String authenticatedUser = "user1";
+    Authentication.setAuthenticatedUserId(authenticatedUser);
+    ProcessInstance expectedProcessInstance = runtimeService.startProcessInstanceByKey("oneTaskProcess");
```

Comment:
I think you should start a second instance of the process using another user, so you make sure the query is actually filtering the results and not returning all of them.

Reply:
The query uses `singleResult` so it should return only one result otherwise it will fail. Does it make sense?

Reply:
Ok I think I got your point now :) I'll start another process instance using a different user.

Reply:
Exactly, in this way, you have two instances in the database, but one is not matching the filter and so it should not be returned.